L1Trigger/Phase2L1ParticleFlow/src/L1TCorrelatorLayer1PatternFileWriter.cc

```diff
@@ -0,0 +1,355 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/L1TCorrelatorLayer1PatternFileWriter.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include <iostream>
+
+L1TCorrelatorLayer1PatternFileWriter::L1TCorrelatorLayer1PatternFileWriter(const edm::ParameterSet& iConfig,
+                                                                           const l1ct::Event& eventTemplate)
+    : partition_(parsePartition(iConfig.getParameter<std::string>("partition"))),
+      writeInputs_(iConfig.existsAs<std::string>("inputFileName") &&
+                   !iConfig.getParameter<std::string>("inputFileName").empty()),
+      writeOutputs_(iConfig.existsAs<std::string>("outputFileName") &&
+                    !iConfig.getParameter<std::string>("outputFileName").empty()),
+      outputBoard_(-1),
+      outputLinkEgamma_(-1),
```

Comment:
Do you actually want these to be always constant? If so I would move their initialization to the class header (and put them as `constexpr`). If not and you want to have a default value, a `fillDescription` may be added to the class. You can do it starting from a an existing `cfi` file and following [this](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Converting_an_Existing_CFI_file).

Reply:
@gpetruc Can you comment?

Reply:
No, they're not always constant, they're filled later in the constructor depending on which detector region is implemented.

A fillDescriptions could be written, but as you can see from the code below in the constructor it will end end up probably being pretty complex as the set of parameters needed depends on the values of the parameters.